packages/request-handler/src/stock-price.ts

```diff
@@ -0,0 +1,52 @@
+import dayjs from "dayjs";
+import { z } from "zod";
+
+import { fetchWithTimeout } from "@homarr/common";
+
+import { createCachedWidgetRequestHandler } from "./lib/cached-widget-request-handler";
+
+export const fetchStockPriceHandler = createCachedWidgetRequestHandler({
+  queryKey: "fetchStockPriceResult",
+  widgetKind: "stockPrice",
+  async requestAsync(input: { stock: string; timeRange: string; timeInterval: string }) {
+    const response = await fetchWithTimeout(
+      `https://query1.finance.yahoo.com/v8/finance/chart/${input.stock}?range=${input.timeRange}&interval=${input.timeInterval}`,
+    );
+    const data = dataSchema.parse(await response.json());
+
+    if ("error" in data) {
+      throw new Error(data.error.description);
+    }
+
+    return responseSchema.parse(data.chart.result[0]);
```

Comment:
This parse is unnecessary, but you'll also have to make the array check to have a length of more than 1